docs/_docs/03-ug-mutable-dataset.md

```diff
@@ -87,7 +89,84 @@ hs.refreshIndex("empIndex", "incremental")
 ```
 
 ### Optimize Index
-TODO
+The fastest way to index newly added source data files and merge them into an existing index is by calling `refreshIndex`
+with the `"incremental"` mode. In this mode, each time refresh is called, it creates fresh index files for new data files
+and updates index metadata to include them in the index content.
+As these index files accumulate, they could affect query performance when the index is used.
```

Comment:
BTW, RefreshDeleteAction will merge them; so append-only incremental refresh ops will make that fragments. I think this could be described in design doc.

Reply:
Yes, I made a bit of change in the text to emphasize more that index files for appended data cause the issue. I think we should not use `RefreshDeleteAction` term in the doc as that is an internal Action. From a user's perspective, it is only `refreshIndex` API. 
@sezruby Let me know if you still think more context is required here.

Reply:
Yes this might be enough for user document.